compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2680,7 +2680,7 @@ OMR::Z::TreeEvaluator::dselectEvaluator(TR::Node *node, TR::CodeGenerator *cg)
       generateRRInstruction(cg, TR::InstOpCode::LDGR, node, tempReg, conditionReg);
       // generate compare with zero
       generateVRIaInstruction(cg, TR::InstOpCode::VGBM, node, vzeroReg, 0, 0);
-      generateVRRcInstruction(cg, TR::InstOpCode::VFCE, node, vectorSelReg, tempReg, vzeroReg, 1, 0, 3);
+      generateVRRcInstruction(cg, TR::InstOpCode::VFCE, node, vectorSelReg, tempReg, vzeroReg, 0, 0, 2);
```

Comment:
This is incorrect. You are basically forcing the VFCE instruction with short-format (Float) so it will generate functionally incorrect code for Double type.

You can use [getVectorElementSizeMask(TR::Node *node)](https://github.com/eclipse/omr/blob/4aa604baf16d8ef55fb13c3b5243984e8edf00a1/compiler/z/codegen/OMRTreeEvaluator.cpp#L15929) to get `m4` mask. 

Also given that operation should only take place for first element, I wonder if we should set in the `m5` to 0x8 which would control the operation to happen only on 0th element , which will be FPR value loaded in `LDGR`.